protocol.md

```diff
@@ -0,0 +1,196 @@
+Status Secure And Decentralized Messaging Protocol
+==================================================
+
+# Abstract
+
+Ethereum empowers users and developers to interact with totally new kind of applications called Dapps (Decentralized applications). These application allows to interact with the blockchain on a completely new level which is not only about exchanging values but also executing arbitrary logic. This logic can form very sophisticated programs like DAOs (Decentralized autonomous organizations). The missing part here is how users of Dapps can communicate securely and in a decentralized way with each other. Communication is an essential part of any activity. In this document, we specify a secure and decentralized messaging protocol that is capable of running on the Ethereum network.
+
+# Introduction
+
+TBD
+
+# Terminology
+
+* *Client*: a Whisper node implementing the protocol
+* *Whisper node*: an Ethereum node with Whisper V6 enabled (in the case of geth, it's `--shh` option)
+* *MailServer*: an Ethereum node with Whisper V6 enabled and a mail server registered capable of storing and providing offline messages
+* *Message*: decrypted Whisper message
+* *Envelope*: encrypted message with some metadata like topic and TTL echanged between Whisper nodes; a symmetric or assymetric key is needed to decrypt it and read the payload
+* *Offline message*: an expired envelope stored by a Whisper node permanently
```

Comment:
I would also add info about a Fleet there.

Reply:
I don't mention fleet below in the spec I believe so that's why I didn't put it in there.

Reply:
I'd explained it here anyway, fleets are a part of our messaging infrastructure, just like mailservers.

Reply:
I tend to agree with @mandrigin here.  If we use it and we rely on it, we should describe it.

Reply:
Well, it's just our internal way of naming a group of servers. Technically each fleet is a separate Whisper network and this is self explanatory. I don't see a point in introducing a term which can be replaced with something much clearer which is not specific to our implementation. Again, this is protocol spec, it should be separate from the implementation.

Reply:
Are they even really isolated networks? Wouldn't a peer with connectivity on two fleets make them share all messages (unless they are running incompatible versions)

Reply:
Yes, it would but in this case they just stop being separate networks in the discussion. I don't see it being important for the protocol itself. It is agnostic how the network is formed. What I can do is to introduce `Whisper network` term.